src/BlueberryStaking.sol

```diff
@@ -747,24 +731,39 @@ contract BlueberryStaking is
     /**
      * @notice Changes the information for the uniswap V3 pool to fetch the price of BLB
      * @param _uniswapPool The new address of the uniswap pool
+     * @param _stableAsset The new address of the stable asset
      * @param _observationPeriod The new observation period for the uniswap pool
      */
     function setUniswapV3Pool(
         address _uniswapPool,
+        address _stableAsset,
         uint32 _observationPeriod
     ) external onlyOwner {
-        if (_uniswapPool == address(0)) {
+        if (_uniswapPool == address(0) || _stableAsset == address(0)) {
             revert AddressZero();
         }
         if (_observationPeriod == 0 || _observationPeriod > 432_000) {
             revert InvalidObservationTime();
         }
+        
+        address token0 = IUniswapV3Pool(_uniswapPool).token0();
+        address token1 = IUniswapV3Pool(_uniswapPool).token1();
+
+        if (_stableAsset != token0 && _stableAsset != token1) {
+            revert InvalidStableAsset();
+        }
```

Comment:
```suggestion
bool blbIsToken0 = IUniswapV3Pool(_uniswapPool).token0() == address(blb)

If (blbIsToken0) {
    address token1 = IUniswapV3Pool(_uniswapPool).token1();
    If (token1 != _stableAsset) revert InvalidStableAsset();
    stableAsset = token1
 } else {
    address token0 = IUniswapV3Pool(_uniswapPool).token0();
    If (token0 != _stableAsset) revert InvalidStableAsset();
    stableAsset = token0
}
```

We can optimize this future will less external calls and comparisons and then reuse blbIsToken0 when setting `uniswapV3Info`.